### PR TITLE
Delete code that is no longer required

### PR DESCRIPTION
**Pull request by rossf7**
Towards https://github.com/giantswarm/giantswarm/issues/20355

This is preparation for removing the creation of the cluster namespace in #148 

The changes are big so this PR just deletes the code we don't need. I'll do another PR with the config and wiring and the final PR with the logic. 

Disclaimer: More PRs may be needed :)

I've commented the PR with why stuff gets deleted.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -14,9 +14,6 @@ require (
 	github.com/giantswarm/microkit v1.0.0
 	github.com/giantswarm/micrologger v0.6.0
 	github.com/giantswarm/operatorkit/v6 v6.1.0
-	github.com/giantswarm/release-operator/v2 v2.5.0
```
Comment (rossf7) [inline go.mod:17]: No more release CRs 🔥

File: go.mod [hunk 2]
```diff
@@ -14,9 +14,6 @@ require (
...
-	github.com/giantswarm/resource/v4 v4.0.0
```
Comment (rossf7) [inline go.mod:18]: I've rewritten the `app` resource to use the simple interface not the CRUD interface. This means we can deprecate the resource library as its only used by cluster-operator.

This PR deletes the app resource and the next will use the simple interface. So the diff is a bit cleaner.

File: helm/cluster-apps-operator/values.yaml [hunk 3]
```diff
@@ -41,29 +41,5 @@ registry:
...
-        cloud-provider-openstack:
-          hasClusterValuesSecret: true
-        # Upgrade force is disabled to avoid affecting customer workloads.
-        coredns:
-          useUpgradeForce: false
-        net-exporter:
-          chart: "net-exporter"
-        # Upgrade force is disabled to avoid affecting customer workloads.
-        nginx-ingress-controller:
-          useUpgradeForce: false
```
Comment (rossf7) [inline helm/cluster-apps-operator/values.yaml:66]: This config all moves to default-apps-openstack.

File: service/controller/cluster.go [hunk 4]
```diff
@@ -243,23 +161,7 @@ func newClusterResources(config ClusterConfig) ([]resource.Interface, error) {
...
-		clusterNamespaceResource, err = clusternamespace.New(c)
-		if err != nil {
-			return nil, microerror.Mask(err)
-		}
-	}
-
 	resources := []resource.Interface{
-		// clusterNamespace manages the namespace for storing app related
-		// resources for this cluster.
-		clusterNamespaceResource,
```
Comment (rossf7) [inline service/controller/cluster.go:262]: The namespace isn't created and we use the org namespace instead.

File: service/controller/cluster.go [hunk 5]
```diff
@@ -271,18 +173,6 @@ func newClusterResources(config ClusterConfig) ([]resource.Interface, error) {
 		resources = append(resources, clusterSecretResource)
 	}
 
-	resources = append(resources,
-		// appResource manages the per cluster app-operator instance and the
-		// workload cluster apps.
-		appResource,
-		// appFinalizerResource removes finalizers after the per cluster
-		// app-operator instance has been deleted.
-		appFinalizerResource,
```
Comment (rossf7) [inline service/controller/cluster.go:280]: Deletion logic changes so we don't delete app-operator until all app CRs are deleted.

**Comment by tfussell** [comment]
This just leaves the cluster-values CM and Secret resources so the next PR will add a new resource which only installs app-operator I guess?

**Review by tfussell — APPROVED** [review]
LGTM. Can't beat getting rid of 2000 LOC.

**Pull request merged**